src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -762,8 +857,40 @@ internal override void Complete()
                     }
 
                     this.Command.Context.SetVariable(SpecialVariables.LastExitCodeVarPath, _nativeProcess.ExitCode);
-                    if (_nativeProcess.ExitCode != 0)
-                        this.commandRuntime.PipelineProcessor.ExecutionFailed = true;
+                    if (_nativeProcess.ExitCode == 0)
+                    {
+                        return;
+                    }
+
+                    this.commandRuntime.PipelineProcessor.ExecutionFailed = true;
+
+                    if (!ExperimentalFeature.IsEnabled(ExperimentalFeature.PSNativeCommandErrorActionPreferenceFeatureName)
+                        || !(bool)Command.Context.GetVariableValue(SpecialVariables.PSNativeCommandUseErrorActionPreferenceVarPath))
```

Comment:
This variable may be deleted by a user in an extreme situation (rare), and in that case, you will get `null` from `GetVariableValue` call, and the cast will fail.

Maybe use the other overload `GetVariableValue(VariablePath path, object defaultValue)`, providing the default value to be `false`.